SKNIBot/Core/Commands/ModerationCommands/StatsCommand.cs

```diff
@@ -1,181 +1,381 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.Diagnostics;
 using System.Linq;
 using System.Text;
+using System.Threading;
 using System.Threading.Tasks;
 using DSharpPlus;
 using DSharpPlus.CommandsNext;
 using DSharpPlus.CommandsNext.Attributes;
 using DSharpPlus.Entities;
 using DSharpPlus.Exceptions;
+using SKNIBot.Core.Database.Models.DynamicDB;
 using SKNIBot.Core.Helpers;
+using SKNIBot.Core.Services.UserMessageStatsService;
+using SKNIBot.Core.Services.DateMessageStatsService;
 
 namespace SKNIBot.Core.Commands.ModerationCommands
 {
     [CommandsGroup("Moderacja")]
-    public class MessageStatsCommand : BaseCommandModule
+    class MessageStatsCommand : BaseCommandModule
     {
         private const int UsernameFieldLength = 40;
         private const int MessagesCountFieldLength = 20;
+        private const int CacheReloadInterval = 1000 * 60 * 60 * 24;
 
         private int TotalFieldsLength => UsernameFieldLength + MessagesCountFieldLength;
 
+        private UserMessageStatsService _statsHookUser;
+        private DateMessageStatsService _statsHookDate;
+        private Timer _cacheReloadTimer;
+
+        public MessageStatsCommand(UserMessageStatsService userMessageStats, DateMessageStatsService dateMessageStats)
+        {
+            _statsHookUser = userMessageStats;
+            _statsHookDate = dateMessageStats;
+            _cacheReloadTimer = new Timer(CacheReloadCallback, null, 40000, Timeout.Infinite);
+        }
+
+        private async void CacheReloadCallback(object state)
+        {
+            _cacheReloadTimer.Change(CacheReloadInterval, Timeout.Infinite);
+            Console.Out.WriteLine("Starting full stats cache reload... This will take a lot of time...");
+            var servers = _statsHookUser.GetServerList();
+            servers.UnionWith(_statsHookDate.GetServerList());
+            foreach (var server in servers)
```

Comment:
What if database has ServerId where bot is no longer present? We don't delete old servers from database. Won't this cause an exception?
And maybe we should reload cache only for server where command was executed.

Reply:
Fixed by applying exception handling with cache invalidation